### PR TITLE
[BUGFIX] Fix Data Preprocessing for Translation Data

### LINKED ISSUE #544: Transformer BLEU of the current code base is worse than previous one
I recently retrained the transformer, and found that it will diverge with the default setting. The divergence issue is caused by use_layer_norm_before_dropout=True and is fixed in #543. However, the final testing BLEU is still worse than the previous result. Compare to the version in #413, the testing BLEU is about 0.8 worse.

### PR DESCRIPTION
**Pull request by szhengac**
## Description ##
Fix #544

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Feature1, tests, (and when applicable, API doc)
- [x] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-568/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-568/2/index.html

File: scripts/machine_translation/dataprocessor.py [hunk 1]
```diff
@@ -86,11 +86,11 @@ def __init__(self, src_vocab, tgt_vocab, src_max_len=None, tgt_max_len=None):
         self._tgt_max_len = tgt_max_len
 
     def __call__(self, src, tgt):
-        if self._src_max_len:
+        if self._src_max_len > 0:
```
Comment (sxjscience) [inline scripts/machine_translation/dataprocessor.py]: Should it be `>=0`? In that case, `src_max_len=0` will trigger an error.

**Comment by eric-haibin-lin** [comment]
Is the accuracy now back to normal with this fix?

**Comment by sxjscience** [comment]
CI failed because https://github.com/facebookresearch/fastText/blob/master/pretrained-vectors.md is not found.

**Comment by szha** [comment]
I fixed that in #574

**Comment by szhengac** [comment]
> Is the accuracy now back to normal with this fix?

Yes.

**Comment by szha** [comment]
@szhengac could you do a rebase?

**Comment by eric-haibin-lin** [comment]
could you take a look at the test failure?

**Pull request merged**